### DISCUSSION TITLE
How to set a transparent color to an object?

### DISCUSSION THREAD

**Discussion by omerxx** [charmbracelet/lipgloss#219] [Q&A]
Hi!
I've tried hiding borders (or other objects for that matter) on certain events, but I can't find the terminal's background color.
`NoColor` doesn't help here as it goes directly to the foreground default which in my case is white.
I don't want to not render the objects as the layout breaks if I do and it makes sense to me to keep them hidden.

Is there a way to access the terminal's background color from lipgloss?

Thanks!

**Comment by meowgorithm** [comment]
Hi! Lip Gloss doesn't expose a way to read the terminal's background color, however you can make the border and background transparent like so:

```go
s := style.
    UnsetBackground()                  // remove the background, making it transparent
    BorderStyle(lipgloss.HiddenBorder) // transparent borders
```

To make the foreground transparent, you could technically replace the characters in the string with spaces…

```go
str := strings.Repeat(" ", lipgloss.Width("Hello! 你好!"))
output := style.Render(str)
```

…however perhaps a helper function would make sense here.

**Comment by beltagyy** [comment]
Lipgloss cannot read the terminal background color. The ANSI protocol has an OSC 11 escape sequence to query it, but support is spotty across terminal emulators and Go TTY libraries, so lipgloss does not implement it.

NoColor sets the ANSI default color, not the actual terminal background, which is why it appears white in your setup.

Workarounds:
- Explicitly set a background color in your lipgloss style that matches your terminal theme.
- Use AdaptiveColor with light/dark variants so your UI looks correct on both dark and light terminals without needing the exact background value.
- If you only target one terminal emulator, you can query OSC 11 yourself and pass the result into lipgloss as an explicit color.
